c10/core/Scalar.h

```diff
@@ -191,11 +191,17 @@ class C10_API Scalar {
   isIntegral() const {
     return Tag::HAS_i == tag || Tag::HAS_si == tag || Tag::HAS_u == tag;
   }
+
   bool isIntegral(bool includeBool) const {
     return Tag::HAS_i == tag || Tag::HAS_si == tag || Tag::HAS_u == tag ||
         (includeBool && isBoolean());
   }
 
+  // See Note [Meaning of HAS_u]
+  bool isUnsigned() const {
+    return Tag::HAS_u == tag || (Tag::HAS_i == tag && v.i >= 0);
+  }
```

Comment:
Yes this is fine. But you might want an alternative that tells you if it fits in 64-bit int or not, since that's what you are probably actually interested in when you do the IValue.

Reply:
That's kind of my debate: should I create some sort of semantically weird `is64bitUnsigned` API here, or check it on the caller side. I'm leaning towards later...